packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx

```diff
@@ -1,113 +1,108 @@
-
-    const { palette } = theme!;
-    this._colors = this.props.colors || [palette.blueLight, palette.blue, palette.blueMid, palette.red, palette.black];
+    const { theme, className, styles, width, barHeight, isMultiStackedBarChart } = this.props;
     this._classNames = getClassNames(styles!, {
       theme: theme!,
       width: width!,
       barHeight: barHeight!,
+      isMultiStackedBarChart: isMultiStackedBarChart,
       className
     });
   }
 
-  private _createBars(data: IDataPoint[], width: number, barHeight: number): JSX.Element[] {
-    let prevWidth = 0;
-    const barWidths = [0];
-    const total = data.reduce((acc: number, point: IDataPoint) => acc + point.y, 0);
-    const bars = data.map((point: IDataPoint, index: number) => {
-      const value = (point.y / total) * width;
-      prevWidth = prevWidth + value;
-      barWidths.push(prevWidth);
-
+  private _createBarsAndLegends(data: IChartProps, barHeight: number, palette: IPalette): [JSX.Element[], JSX.Element] {
+    const defaultPalette: string[] = [palette.blueLight, palette.blue, palette.blueMid, palette.red, palette.black];
+    const legendDataItems: ILegend[] = [];
+    // calculating starting point of each bar and it's range
+    const startingPoint: number[] = [];
+    const total = data.chartData!.reduce(
+      (acc: number, point: IChartDataPoint) => acc + (point.data ? point.data : 0),
+      0
+    );
+    let prevPosition = 0;
+    let value = 0;
+    const bars = data.chartData!.map((point: IChartDataPoint, index: number) => {
+      const color: string = point.color ? point.color : defaultPalette[Math.floor(Math.random() * 4 + 1)];
+      const pointData = point.data ? point.data : 0;
+      // mapping data to the format Legends component needs
+      const legend: ILegend = {
+        title: point.legend!,
+        color: color,
+        action: this._legendClickAction
+      };
+      legendDataItems.push(legend);
+      if (index > 0) {
+        prevPosition += value;
+      }
+      value = (pointData / total) * 100;
+      startingPoint.push(prevPosition);
       return (
-        <rect
-          key={index}
-          x={barWidths[index]}
-          y={0}
-          width={value}
-          height={barHeight}
-          fill={this._colors[index % this._colors.length]}
-        />
+        <rect key={index} x={startingPoint[index] + '%'} y={0} width={value + '%'} height={barHeight} fill={color} />
       );
     });
-    return bars;
+    const legends = <Legends legends={legendDataItems} />;
+    return [bars, legends];
   }
 
-  private _createLegendBars(data: IDataPoint[]): JSX.Element {
-    const legendDataItems: ILegendDataItem[] = data.map((point: IDataPoint, index: number) => {
-      return {
-        legendText: point.x,
-        legendColor: this._colors[index % this._colors.length]
-      };
-    });
-
-    return <Legend renderData={legendDataItems} />;
+  private _legendClickAction(): void {
+    alert('legend clicked');
```

Comment:
> alert('legend clicked'); [](start = 3, length = 25)

do we need this?

Reply:
removed it, was placed as stub for hovercard behaviour we are going to implement next